### DISCUSSION TITLE
Is it possible to run local test without installing tensorflow?

### DISCUSSION THREAD

**Discussion by chrjx** [photoprism/photoprism#1802] [Q&A] [answered]
Hi everyone, 

I want to run unit test on my laptop (Apple M1). When I type `go test ./internal/config -v`, it shows:
```
# github.com/tensorflow/tensorflow/tensorflow/go
ld: library not found for -ltensorflow
clang: error: linker command failed with exit code 1 (use -v to see invocation)
FAIL    github.com/photoprism/photoprism/internal/config [build failed]
FAIL
```

I wonder if I can skip tensorflow part and just run some simple tests?

**Comment by lastzero** [comment, marked as answer]
In theory yes, e.g. with compiler flags. Not sure if those would work / be enough since the package is pulled before building via [go.mod](https://github.com/photoprism/photoprism/blob/develop/go.mod). You're welcome to perform research on this!

It's all a matter of work / time.
